### PR TITLE
fix: making lefthook working, removing commitLint workflow [KHCP-8829]

### PR DESCRIPTION
**Pull request by ValeryG**
# Summary

- renaming `lefthook.yml to `lefthook.yaml` so that `yarn lefthook install` picks it up
- removing commitLint workflow

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 # Workflows & CI
 /.github/ @adamdehaven @jillztom
-/lefthook.yml @adamdehaven @jillztom
+/lefthook.yaml @adamdehaven @jillztom
```
Comment (mihai-peteu) [inline .github/CODEOWNERS:8]: 🙃

**Review by mihai-peteu — APPROVED** [review]

File: .github/workflows/commitlint.yml [hunk 2]
```diff
@@ -1,14 +0,0 @@
-name: Lint Commit Messages
```
Comment (adamdehaven) [inline .github/workflows/commitlint.yml:1]: Why is this being removed? This enforces our commit hooks that are required for semantic release

Comment (ValeryG) [reply .github/workflows/commitlint.yml:1]: This is all in JIRA and link to the slack thread. commitLints are part of the pre-commit hocks and enabled in left hook (wasn't working due to `.yaml`). This is the way we are doing things in other repos (public-ui-components), unification is good. listing commit messages after commit happens is too late in the game...

Comment (adamdehaven) [reply .github/workflows/commitlint.yml:1]: So what happens if someone skips the pre-commit hooks when they push their commits? I don't see the harm in leaving the check in place

**Review by kaiarrowood — APPROVED** [review]

File: .github/workflows/commitlint.yml (hunk 2 above)
Comment (ValeryG) [reply .github/workflows/commitlint.yml:1]: same thing what happens in public-ui-components and other repos. If we need  keep  this workflow in this repo but not in any other public repos - we need a better reason and / or comment in this workflow as of why it so unique and why it needs to be there. Another reason to drop it - those failures we see for 2 weeks. https://github.com/Kong/kongponents/actions/workflows/commitlint.yml?page=2 - those are fixable but there is no reason to fix it. remove workflow is cleaner.

**Review by jillztom — APPROVED** [review]

**Comment by adamdehaven** [comment]
@jillztom there is now nothing in this repository enforcing the commit format, as git hooks can easily be skipped. 

I’m very opposed as to why this PR was merged.

**Comment by jillztom** [comment]
> @jillztom there is now nothing in this repository enforcing the commit format, as git hooks can easily be skipped.
> 
> I’m very opposed as to why this PR was merged.

Let's discuss this when you get back.

**Pull request merged**